drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/test/java/org/drools/workbench/screens/guided/dtable/client/wizard/column/pages/FieldPageTest.java

```diff
@@ -374,22 +374,75 @@ public void testAsWidget() {
     }
 
     @Test
-    public void testSetupField() {
+    public void testSetupFieldWhenConstraintValueIsNotPredicate() {
         final String factField = "factField";
 
+        doReturn(false).when(page).isConstraintValuePredicate();
         doReturn(factField).when(page).getFactField();
 
-        doNothing().when(page).forEachFactField(this.consumer.capture());
+        doNothing().when(page).forEachFactField(consumer.capture());
 
         page.setupField();
 
-        final Consumer<String> consumer = this.consumer.getValue();
-
-        consumer.accept(factField);
+        consumer.getValue().accept(factField);
 
-        verify(view).addItem(factField,
-                             factField);
-        verify(view).setupFieldList();
+        verify(view).enableFieldList();
+        verify(view).addItem(factField, factField);
+        verify(view).setupEmptyFieldList();
         verify(view).selectField(factField);
     }
+
+    @Test
+    public void testSetupFieldWhenConstraintValueIsPredicate() {
+
+        final String factField = "factField";
+
+        doReturn(true).when(page).isConstraintValuePredicate();
+        doReturn(factField).when(page).getFactField();
+
+        page.setupField();
+
+        view.enableTextField();
+        view.setField(factField);
+    }
+
+    @Test
+    public void testSetupPatternWarningMessagesWhenFactTypeIsNotNil() {
+
+        final String factType = "factType";
+
+        doReturn(factType).when(page).factType();
+
+        page.setupPatternWarningMessages();
+
+        verify(view).patternWarningToggle(false);
+    }
+
+    @Test
+    public void testSetupPatternWarningMessagesWhenFactTypeIsNilAndConstraintValueIsPredicate() {
```

Comment:
@karreiro This test is great, it brought me to this observation:
- If Calculation Type is Predicate, Field Page doesn't force the user to specify pattern
- However, the column wizard always force the user to specify pattern on the pattern page
- Second however, I am not sure, how to solve this

Reply:
A Pattern always needs to be selected; therefore IMO even if the Calculation Type is Predicate the warning about Pattern not being selected should be shown on the Field page. When using a Predicate it is not possible to complete the wizard unless a Pattern has been selected.

Reply:
Thanks Michael, +1 from my site. Guilherme, please incorporate also this.